src/app/plan/components/Wizard/VolumesTable.tsx

```diff
@@ -15,80 +15,56 @@ import Loader from 'react-loader-spinner';
-  state = {
-    options: [{ label: 'copy', value: 'copy' }, { label: 'move', value: 'move' }],
-    selectedOption: null,
-    rows: [],
+  const handleTypeChange = (row, option) => {
+    updateTableData(row.index, option.value);
   };
-  handleTypeChange = (row, option) => {
-    this.updateTableData(row.index, option.value);
-  };
-  getCurrentPlan = (): any => {
-    return this.props.plans.find(p => p.MigPlan.metadata.name === this.props.values.planName);
+
+  const getCurrentPlan = () => {
+    return planList.find(p => p.MigPlan.metadata.name === values.planName);
   };
 
-  updateTableData = (rowIndex, updatedValue) => {
-    const currentPlan = this.getCurrentPlan();
-    const { values } = this.props;
-    const { rows } = this.state;
+  const updateTableData = (rowIndex, updatedValue) => {
+    const currentPlan = getCurrentPlan();
     const rowsCopy = [...rows];
     if (currentPlan !== null && values.persistentVolumes) {
       const updatedRow = { ...rowsCopy[rowIndex], type: updatedValue };
 
       rowsCopy[rowIndex] = updatedRow;
     }
-    this.setState({ rows: rowsCopy });
-    this.props.setFieldValue('persistentVolumes', rowsCopy);
+
+    setRows(rowsCopy);
+    props.setFieldValue('persistentVolumes', rowsCopy);
   };
 
-  getTableData() {
-    const currentPlan = this.getCurrentPlan();
-    // Builds table data from a combination of the formik values, and the
-    // persistent volumes as seen on a MigPlan object.
+  useEffect(() => {
+    const currentPlan = getCurrentPlan();
     if (currentPlan) {
-      const discoveredPersistentVolumes = currentPlan.MigPlan.spec.persistentVolumes || null;
-
-      // No PVs discovered to be in use. This is normal for stateless cloud apps.
-      if (!discoveredPersistentVolumes) {
-        return [];
-      }
+      const discoveredPersistentVolumes = currentPlan.MigPlan.spec.persistentVolumes || [];
       let mappedPVs;
-      if (this.props.values.persistentVolumes) {
+      if (values.persistentVolumes) {
         mappedPVs = discoveredPersistentVolumes.map(planVolume => {
           let pvAction = 'copy'; // Default to copy
-          if (this.props.values.persistentVolumes.length !== 0) {
-            const rowVal = this.props.values.persistentVolumes.find(
-              v => v.name === planVolume.name
-            );
+          if (values.persistentVolumes.length !== 0) {
+            const rowVal = values.persistentVolumes.find(v => v.name === planVolume.name);
             pvAction = rowVal.type;
           }
-
           // TODO: A number of these values will need to be further supported by the
           // controller. Today the data is not available.
           // See the mig controller issue describing what data we need here:
           // https://github.com/fusor/mig-controller/issues/134
           return {
             name: planVolume.name,
             project: '',
-            storageClass: '',
+            storageClass: planVolume.selection.storageClass || '',
```

Comment:
@eriknelson when discovering PVs, there is already a default selection on the PV for storage class taht is added to values.persistentVolumes here. On the next step in the storageClass selection, you can change the object value on the PV to the selection of your choice in the dropdown. see: https://github.com/fusor/mig-ui/pull/364/files#diff-2848a72f04be20e44dd3940a1f78c2b9R29

Reply:
Still need to figure out a way to test multiple storage classes for selection, though.